app/src/main/kotlin/com/wire/android/ui/home/userprofile/UserProfileScreen.kt

```diff
@@ -1,16 +1,343 @@
+            )
+        }
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ColumnScope.renderHeader(state: UserProfileState, viewModel: UserProfileViewModel) {
+    UserProfileAvatar(
+        modifier = Modifier
+            .padding(top = dimensions().spacing16)
+            .align(Alignment.CenterHorizontally),
+        size = dimensions().userAvatarDefaultBigSize,
+        avatarUrl = state.avatarUrl,
+        status = UserStatus.NONE
+    )
+
+    ConstraintLayout(modifier = Modifier.align(Alignment.CenterHorizontally)) {
+
+        val (data, editBtn, team) = createRefs()
+
+        Column(modifier = Modifier
+            .padding(horizontal = dimensions().spacing64)
+            .constrainAs(data) {
+                top.linkTo(parent.top)
+                bottom.linkTo(parent.bottom)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            }) {
+            Text(
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                text = state.fullName,
+                overflow = TextOverflow.Ellipsis,
+                maxLines = 1,
+                style = MaterialTheme.wireTypography.title02,
+                color = MaterialTheme.colorScheme.onBackground,
+            )
+
+            Text(
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                text = state.userName,
+                overflow = TextOverflow.Ellipsis,
+                style = MaterialTheme.wireTypography.body02,
+                maxLines = 1,
+                color = MaterialTheme.colorScheme.onBackground,
+            )
+        }
+
+        IconButton(
+            modifier = Modifier
+                .padding(start = dimensions().spacing16)
+                .constrainAs(editBtn) {
+                    top.linkTo(data.top)
+                    bottom.linkTo(data.bottom)
+                    end.linkTo(data.end)
+                },
+            onClick = { viewModel.editProfile() },
+            content = Icons.Filled.Edit.Icon()
+        )
+
+        Text(
+            modifier = Modifier
+                .padding(top = dimensions().spacing8)
+                .padding(horizontal = dimensions().spacing16)
+                .constrainAs(team) {
+                    top.linkTo(data.bottom)
+                    start.linkTo(parent.start)
+                    end.linkTo(parent.end)
+                },
+            text = state.teamName,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            style = MaterialTheme.wireTypography.label01,
+            color = MaterialTheme.colorScheme.onBackground,
+        )
+    }
+}
+
+@Composable
+private fun ColumnScope.renderStatusesRow(state: UserProfileState) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(dimensions().spacing16),
+    ) {
+        WireSecondaryButton(
+            onClick = {},
```

Comment:
Should we at least link this click event to a common method on the VM where we will define the logic later?